### DISCUSSION TITLE
Status of package: org.opengis.geometry.coordinate

### DISCUSSION THREAD

**Discussion by paulushub** [opengeospatial/geoapi#107] [Q&A] [answered]
The file `/geoapi/src/main/java/org/opengis/geometry/coordinate/package-info.java` is marked as

```java
/**
 * <b>Deprecated</b> — {@code Position} has been removed in ISO 19107:2019.
 *
 * @author  Martin Desruisseaux (IRD, Geomatys)
 * @author  Axel Francois (LSIS/Geomatys)
 * @version 3.1
 * @since   1.0
 */
@Deprecated
package org.opengis.geometry.coordinate;
```

So, is the content of the following **pending** package depreciated?
`org.opengis.geometry.coordinate`

**Comment by desruisseaux** [comment, marked as answer]
The `org.opengis.geometry` packages in the `geoapi-pending` module should be avoided for production codes (usages in experimental codes are appreciated for providing feedbacks, but at high risk). An initial version of these interfaces was based on ISO 19107:2003. Two Ph.D. students, one in Germany and one in France, tried to implement these interfaces. But these efforts did not resulted in production code. Then, ISO 19107 started a major revision cycle, and we started to modify these interfaces following the draft versions of ISO 19107. It was part of the work of the French Ph.D. student, in the goal of providing implementation feedbacks to the ISO 19107 editor. But the student finished his Ph.D. before the final ISO 19107:2019 release, so a lot of additional changes are missing. Therefore, the geometry interfaces in `geoapi-pending` stand somewhere between ISO 19107:2003 and ISO 19107:2019. We did not yet completed the upgrade (that task plus a "proof of concept" implementation could almost be another Ph.D. topic), neither we evaluated the amount of changes that it would imply. The only thing that we completed fully was checking all implications for the non-pending part of GeoAPI, and one of these implications was the deprecation of `Position`.

The `package-info` pointed in this discussion, which is on the `master` branch, is a left-over that we forgot to delete. I just deleted it now. The file is still present, together with the `Position` interface, on the GeoAPI 3.1 branch. The reason is that this interface was present in GeoAPI 3.0 official release (which contained a very small subset of ISO 19107), so we cannot just delete it. GeoAPI 3.1 deprecates it, and GeoAPI 4.0 deletes it.

**Reply by paulushub** [reply]
Thanks for the clarification. I have all along assumed ISO are approved after a verified implementation!

Currently, I cannot afford the ISO specifications, mine is a one-man company and currently not providing GIS services.
I have worked on GIS products in the past and trying to do so in C# for open source, so I could only afford to port the Java/GeoAPI interfaces. So, I am porting both the approved and pending interfaces to see what can be implemented.

**Reply by desruisseaux** [reply]
> I have all along assumed ISO are approved after a verified implementation!

It has not always been the case. On OGC side, it was not really a requirement in 2003. This is a requirement today, but it has been added many years after 2003. On ISO side, I'm not sure what is the current policy.

> So, I am porting both the approved and pending interfaces to see what can be implemented.

Great :-) I suggest to focus on the non-pending part. Everything in pending is risky. Usually, interfaces are in "pending" for three reasons: either we know that they are obsolete compared to more recent OGC/ISO standards (case of `org.opengis.geometry` and most of what has been derived from XML, such as `org.opengis.style`), or that that big changes may be coming (case of `org.opengis.style`), or either they have no satisfying "proof of concept" implementation (case of `org.opengis.geometry`).

Note: issue #35 is about C# interfaces, but got no progress for the last 6 years. You are welcome to report progress on that issue, or create a new issue or discussion, if you wish.

**Reply by paulushub** [reply]
> Great :-) I suggest to focus on the non-pending part. 

It is not a standard GeoAPI effort as I cannot afford that currently, but I need to create many "standard" interfaces for my effort, 
so I have no choice. Moreover, I cannot wait for the standard effort, but anyone with that interest can use my work as a
starting point as it is quite close to the Java version - and I can contribute.
I have ported the `JSR 385` interfaces and will port the `Threeten` for the temporal interfaces and implementations, which is the only thing left before I start cleaning up, improving and testing the interfaces.
Currently, I have about 20 errors to clear, many due to the extensions and default implementations. I am trying to keep things similar to the Java interfaces so that it will be easy to follow the changes/updates.

> You are welcome to report progress on that issue, or create a new issue or discussion, if you wish.

I hope to make the announcement here for anyone interested to take over from there, I do not have access to the paid ISO specifications and cannot say more than I learn from the Java/GeoAPI and other freely available internet sources.

**Reply by desruisseaux** [reply]
> I do not have access to the paid ISO specifications

Just for information, some of those standards are join OGC/ISO standards. Those ones are freely available, with differences only in cover pages and formatting. In particular:

* ISO 19111 content is identical to [OGC Topic 2](https://docs.ogc.org/as/18-005r8/18-005r8.pdf)
* ISO 19143 content is identical to [OGC 09-026](http://docs.opengeospatial.org/is/09-026r2/09-026r2.html)

**Reply by paulushub** [reply]
Thanks so much, for the ISO 19111/OGC Topic 2 in particular. I have the OGC 09-026.
